api/envoy/data/extensions/filter/udp/dns_filter/v2alpha/inline_dns_table.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+
+package envoy.data.extensions.filter.udp.dns_filter.v2alpha;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.data.extensions.filter.udp.dns_filter.v2alpha";
+option java_outer_classname = "InlineDnsTableProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// [#protodoc-title: DNS Filter Table Data]
+// :ref:`DNS Filter config overview <config_udp_listener_filters_dns_filter>`.
+
+// This message contains the configuration for the DNS Filter if populated
+// from the control plane
+message InlineDnsTable {
+  // This message contains a list of IP addresses returned for a query for a known name
+  message AddressList {
+    repeated string address = 1;
+  }
+
+  // This message type is extensible and can contain a list of addresses
+  // or dictate some other method for resolving the addresses for an
+  // endpoint
+  message DnsEndpoint {
+    oneof endpoint_config {
+      AddressList address_list = 1;
+    }
+  }
+
+  message DnsVirtualDomain {
+    // The domain name for which Envoy will respond to query requests
+    string name = 1 [(validate.rules).string = {min_len: 2 well_known_regex: HTTP_HEADER_NAME}];
+
+    // The configuration containing the method to determine the address
+    // of this endpoint
+    DnsEndpoint endpoint = 2;
```

Comment:
What will you do if the number of endpoints exceeds the size of the UDP datagram? Should there be truncation config around random selection, etc.?

Reply:
The filter returns a random address from the set of addresses.   This would achieve some level of load balancing in away reduce client pinning.   Since we're only returning one address we are well under the 512 byte limit.  There are docs that suggest we can return up to 4096 bytes, but I haven't gone very deep into the extensions and indications that clients can support that.

I can certainly add a `TODO` to expand the number of addresses returned once things are working end to end.

Reply:
yeah I think we are going to want to expand this to return multiple results like a normal DNS server does but that can be done later. Either way I would clearly document the current behavior.

Reply:
Ok, I'll spend some time updating the docs.